src/components/com_tjcertificate/media/js/template.js

```diff
@@ -93,5 +94,30 @@ var template = {
 
 			editor.setValue(templateBody);
 		}
+	},
+	
+	renderCustomTemplate: function(templateId) 
+	{
+		var siteRoot = Joomla.getOptions("system.paths").base;
```

Comment:
use promise here similar to `loadDefaultTemplate`.
add a service to load loadCustomTemplate inside tjCertificateService.js

Reply:
done